shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -150,13 +152,15 @@ @implementation FlutterTextInputPlugin {
 }
 
 - (instancetype)initWithViewController:(FlutterViewController*)viewController {
-  self = [super init];
+  // The view needs a non-zero frame.
+  self = [super initWithFrame:NSMakeRect(0, 0, 1, 1)];
   if (self != nil) {
+    _flutterViewController = viewController;
     _channel = [FlutterMethodChannel methodChannelWithName:kTextInputChannel
                                            binaryMessenger:viewController.engine.binaryMessenger
                                                      codec:[FlutterJSONMethodCodec sharedInstance]];
     _shown = FALSE;
-    __weak FlutterTextInputPlugin* weakSelf = self;
+    __unsafe_unretained FlutterTextInputPlugin* weakSelf = self;
```

Comment:
This doesn't seem right to me - why aren't we keeping this as `__weak`? Why would we want to continue to address the memory this occupied after it was deallocated?

Reply:
NSTextView does not weak https://stackoverflow.com/questions/35360911/why-cannot-nstextview-for-a-weak-reference
The doc https://developer.apple.com/library/archive/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011226
suggest us to use __unsafe_unretained

Reply:
This is the price we pay by converting the FlutterTextInputPlugin into a subclass of NSTextView so that it can be field editor.

I have thought about create another NSTextView as the field editor, but then we need to sync the text editing event between that field editor and FlutterTextInputPlugin which is a real PAIN

Reply:
That same doc mentions that __unsafe_unretained will leave your pointer dangling. But I guess we think it's safe because  `dealloc` nils out the handler.

I think that makes sense, but we should probably add a comment here explaining that we can't use __weak and that we unregister this callback in dealloc